mentoring/dashboard.py

```diff
@@ -0,0 +1,406 @@
+                        condition, value = line.split(':')
+                        value = value.strip()
+                        if condition.isnumeric():  # A condition just listed as an exact value
+                            condition = "x == " + condition
+                    else:
+                        condition = "1"  # Always true
+                        value = line
+                    rules.append(ColorRule(condition, value))
+                except ValueError:
+                    if ignore_errors:
+                        continue
+                    raise ValueError(
+                        _("Invalid color rule on line {line_number}").format(line_number=lineno + 1)
+                    )
+        return rules
+
+    @lazy
+    def color_rules_parsed(self):
+        """
+        Caching property to get parsed color rules. Returns a list of ColorRule objects.
+        """
+        return self.parse_color_rules_str(self.color_rules) if self.color_rules else []
+
+    def _get_submission_key(self, usage_key):
+        """
+        Given the usage_key of an MCQ block, get the dict key needed to look it up with the
+        submissions API.
+        """
+        return dict(
+            student_id=self.runtime.anonymous_student_id,
+            course_id=unicode(usage_key.course_key),
+            item_id=unicode(usage_key),
+            item_type=usage_key.block_type,
+        )
+
+    def color_for_value(self, value):
+        """ Given a string value, get the color rule that matches, if any """
+        if isinstance(value, basestring):
+            if value.isnumeric():
+                value = float(value)
+            else:
+                return None
+        for rule in self.color_rules_parsed:
+            if rule.matches(value):
+                return rule.color_str
+        return None
+
+    def _get_user_full_name(self):
+        """
+        Get the full name of the current user, for the downloadable report.
+        """
+        user_service = self.runtime.service(self, 'user')
+        if user_service:
+            return user_service.get_current_user().full_name
+        return ""
+
+    def _get_course_name(self):
+        """
+        Get the name of the current course, for the downloadable report.
+        """
+        try:
+            course_key = self.scope_ids.usage_id.course_key
+        except AttributeError:
+            return ""  # We are not in an edX runtime
+        try:
+            course_root_key = course_key.make_usage_key('course', 'course')
+            return self.runtime.get_block(course_root_key).display_name
+        except Exception:  # ItemNotFoundError most likely, but we can't import that exception in non-edX environments
+            # We may be on old mongo:
+            try:
+                course_root_key = course_key.make_usage_key('course', course_key.run)
+                return self.runtime.get_block(course_root_key).display_name
+            except Exception:
```

Comment:
I'm a bit surprised you didn't have to disable Pylint's checking for broad exceptions here.